### ISSUE TITLE
Activating Shield Generator Makes Entire Screen Pink

### ISSUE THREAD

**Issue by MrDolphinCodes** [timkurvers/valheim-macos#95]
Whenever I activate the new shield generator my entire screen become pink (image below) and even if I leave the area of the shield it remains so. I have tried re-logging and it does not fix it either. This bug seems to only be happening on my mac and not my pc. To end up fixing it I had a friend log onto my world on his pc and destroy the shield generator which made it normal again. Not sure if you can do anything about it or if it is just a bug that the Valheim developers need to fix. Either way if anyone comes across this bug just destroy the generator and everything will be fixed.
<img width="1440" alt="Screenshot 2024-04-26 at 12 31 39 AM" src="https://github.com/timkurvers/valheim-macos/assets/106354581/7d0ad73d-14f3-45ee-8744-f27a7c64da09">

**Comment by timkurvers** [comment]
Thanks for reporting this. When testing Ashlands briefly, I noticed a shader error in the log, something along the lines of: `ERROR: Shader: Shield Generator is incompatible with this GPU`.

Preliminary thought: Valheim 0.217.40 seemingly [introduced some new features for modding](https://www.valheimgame.com/support/modding-faq-for-the-asset-bundle-update-0-217-40/), so perhaps we can leverage that to 'fix' this shield generator shader so that it is compatible with OpenGL 4.1 on macOS.

**Comment by MrDolphinCodes** [comment]
I'll take a look at trying to make a client side mod that'll fix the issue but I don't have much modding experience other than Minecraft. Also I am going to see if I can get it to work using Apple's [Game Porting Toolkit](https://www.applegamingwiki.com/wiki/Game_Porting_Toolkit). I have been able to get Elden Ring and several other games running smoothly on my M1 Macbook Air using this method.

**Comment by timkurvers** [comment]
> Also I am going to see if I can get it to work using Apple's [Game Porting Toolkit](https://www.applegamingwiki.com/wiki/Game_Porting_Toolkit). I have been able to get Elden Ring and several other games running smoothly on my M1 Macbook Air using this method.

I can confirm the game runs via CrossOver 24.x (which contains parts of Apple's Game Porting Toolkit) if that is of any help. Would expect the toolkit standalone to also run the game.

**Comment by MrDolphinCodes** [comment]
The shield generator is working on the Game Porting Toolkit as well as CrossOver so this is a workaround option for now. It isn't the greatest solution as the game gets a bit choppy and slow unlike your port which has it running like butter. 
<img width="1440" alt="Screenshot 2024-04-26 at 12 44 11 PM" src="https://github.com/timkurvers/valheim-macos/assets/106354581/b330fbb1-ab0f-4b1e-9d4c-973f310dafbf">

**Comment by drummercraig** [comment]
I have run into the pink issue by dying in Lava. As soon as I get close to retrieving my tombstone it turns pink again and cannot do anything. Just wanted to add this as it's not just Shield Generator problem.

**Comment by MrDolphinCodes** [comment]
Very strange. I have beaten the ashlands in several playthroughs (I have practically explored the entire biome on one of them in creative) and haven’t run into that issue at all other than the generator. Is this on multiplayer or single player and is there anything in the area that you suspect might be causing this issue?

**Comment by drummercraig** [comment]
Yes. I've played a bit in Ashlands, raided castles etc. Could you guide me to where the log might be or what it's called?
Also, I am playing on a modded world and that may be part of it...although nothing was touched in the Ashlands.

**Comment by timkurvers** [comment]
> Could you guide me to where the log might be or what it's called?

You can find the log files here: `~/Library/Logs/IronGate/Valheim`, where `~` is your home folder. If you cannot see the Library folder there, you may have to enable it using View Options in your home folder.

**Comment by drummercraig** [comment]
Well I wasn't able to duplicate the issue dying in Lava again but apparently I did go into an area that had a shield generator as it went pink. Here is an excerpt of the log:

[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ShieldGenerator.UpdateShield () (at <186f1bc3e8634801b40e626890156b17>:0)

NullReferenceException: Object reference not set to an instance of an object
  at ShieldGenerator.UpdateShield () [0x000e8] in <186f1bc3e8634801b40e626890156b17>:0 

[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ShieldGenerator.UpdateShield () (at <186f1bc3e8634801b40e626890156b17>:0)

NullReferenceException: Object reference not set to an instance of an object
  at ShieldGenerator.UpdateShield () [0x000e8] in <186f1bc3e8634801b40e626890156b17>:0 

[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ShieldGenerator.UpdateShield () (at <186f1bc3e8634801b40e626890156b17>:0)

NullReferenceException: Object reference not set to an instance of an object
  at ShieldGenerator.UpdateShield () [0x000e8] in <186f1bc3e8634801b40e626890156b17>:0 

WARNING: RGBA Compressed BC7 sRGB format is not supported, decompressing texture
WARNING: RGBA Compressed BC7 UNorm format is not supported, decompressing texture
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ShieldGenerator.UpdateShield () (at <186f1bc3e8634801b40e626890156b17>:0)

NullReferenceException: Object reference not set to an instance of an object
  at ShieldGenerator.UpdateShield () [0x000e8] in <186f1bc3e8634801b40e626890156b17>:0 

[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ShieldGenerator.UpdateShield () (at <186f1bc3e8634801b40e626890156b17>:0)

NullReferenceException: Object reference not set to an instance of an object
  at ShieldGenerator.UpdateShield () [0x000e8] in <186f1bc3e8634801b40e626890156b17>:0 

WARNING: RGBA Compressed BC7 UNorm format is not supported, decompressing texture
WARNING: RGBA Compressed BC7 sRGB format is not supported, decompressing texture
WARNING: RGBA Compressed BC7 UNorm format is not supported, decompressing texture
WARNING: RGBA Compressed BC7 sRGB format is not supported, decompressing texture
WARNING: RGBA Compressed BC7 sRGB format is not supported, decompressing texture
WARNING: RGBA Compressed BC7 sRGB format is not supported, decompressing texture
WARNING: RGBA Compressed BC7 UNorm format is not supported, decompressing texture
WARNING: RGBA Compressed BC7 sRGB format is not supported, decompressing texture
WARNING: RGBA Compressed BC7 UNorm format is not supported, decompressing texture
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ShieldGenerator.UpdateShield () (at <186f1bc3e8634801b40e626890156b17>:0)

**Comment by Dm1tryRin** [comment]
The pink colour in Unity is usually materials that don't fit the right Render Pipeline.

**Comment by timkurvers** [comment]
Just a quick update: contacted IronGate about this and got the following response:

> There will be more compatibility introduced in near future.

There is hope 🙌

**Comment by timkurvers** [comment]
Closing this issue as the official macOS client is now available! 🥳 (see #104)
